lib/resque/worker.rb

```diff
@@ -484,21 +484,27 @@ def self.all_heartbeats
     def self.all_workers_with_expired_heartbeats
       workers = Worker.all
       heartbeats = Worker.all_heartbeats
+      now = server_time
 
       workers.select do |worker|
         id = worker.to_s
         heartbeat = heartbeats[id]
 
         if heartbeat
-          seconds_since_heartbeat = (Time.now - Time.parse(heartbeat)).to_i
+          seconds_since_heartbeat = (now - Time.parse(heartbeat)).to_i
           seconds_since_heartbeat > Resque.prune_interval
         else
           false
         end
       end
     end
 
-    def heartbeat!(time = Time.now)
+    def self.server_time
+      time, _ = redis.time
```

Comment:
I think access to redis is supposed to be done through `data_store` now, with Resque::DataStore delegating to redis